scripts/remove_unnecessary_versions_of_repositories.py

```diff
@@ -0,0 +1,181 @@
+        params["retain_repo_versions"],
+    )
+    logger.debug(result)
+    params = {"retain_repo_versions": pulp_repo["retain_repo_versions"]}
+    result = await pulp_client.request("PATCH", endpoint, json=params)
+    logger.debug(
+        "Create task to back retain repo versions to %s",
+        params["retain_repo_versions"]
+    )
+    logger.debug(result)
+
+
+async def get_all_pulp_repositories(
+    pulp_client: PulpClient,
+    logger: logging.Logger,
+):
+    endpoint = "pulp/api/v3/repositories/rpm/rpm/"
+    params = {"fields": "id,pulp_href,retain_repo_versions"}
+    result = []
+
+    def update_result(response: typing.List[dict]):
+        for rec in response:
+            result.append(rec)
+
+    response = await pulp_client.request("GET", endpoint, params=params)
+    update_result(response.get("results", []))
+    next_page = response.get("next")
+    if not next_page:
+        return result
+    while True:
+        if (
+            "limit" in next_page
+            and re.search(r"limit=(\d+)", next_page).groups()[0] == "100"
+        ):
+            next_page = next_page.replace("limit=100", "limit=1000")
+        parsed_url = urllib.parse.urlsplit(next_page)
+        path = parsed_url.path + "?" + parsed_url.query
+        response = await pulp_client.get_by_href(path)
+        update_result(response.get("results", []))
+        next_page = response.get("next")
+        if not next_page:
+            break
+    logger.debug("%s repositories received from Pulp", len(result))
+    return result
+
+
+async def get_production_repositories(
+    pulp_client: PulpClient,
+    logger: logging.Logger,
+):
+    async with database.Session() as db:
+        repo_q = select(models.Repository).where(
+            models.Repository.production == True
+        )
+        repositories = await db.execute(repo_q)
+    result = repositories.scalars().all()
+    logger.debug("%s production repositories received from DB", len(result))
+    return result
+
+
+async def main():
+    pulp_host = os.environ["PULP_HOST"]
+    pulp_user = os.environ["PULP_USER"]
+    pulp_password = os.environ["PULP_PASSWORD"]
+    args = parse_args()
+    logger = logging.getLogger("pulp_artifacts_remover")
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.INFO)
+    pulp_client = PulpClient(pulp_host, pulp_user, pulp_password)
+
+    logger.info("Get all pulp repos")
+    pulp_repos = await get_all_pulp_repositories(pulp_client, logger)
+    logger.info("Get all production repos")
+    prod_repos = await get_production_repositories(pulp_client, logger)
+    prod_repos_href = []
+    for prod_repo in prod_repos:
+        prod_repos_href.append(prod_repo.pulp_href)
+    logger.info("Removing unnecessery versions of repositories...")
+    for pulp_repo in pulp_repos:
+        if pulp_repo["pulp_href"] in prod_repos_href:
+            retain_repo_versions = 3
```

Comment:
I suggest to leave 3 versions for all RPM repos. That should leave enough data to be able to track all needed info for SBOM